main/acle.rst

```diff
@@ -2226,6 +2226,201 @@ or compiler-generated functions.
+   +--------------+-------------------------------------+---------------------+-------------------------------------+
+   | 40           | ``FEAT_FHM``                        | fhm                 | ID_AA64ISAR0_EL1.FHM == 0b0001      |
+   +--------------+-------------------------------------+---------------------+-------------------------------------+
+   | 50           | ``FEAT_DotProd``                    | dotprod             | ID_AA64ISAR0_EL1.DP == 0b0001       |
+   +--------------+-------------------------------------+---------------------+-------------------------------------+
+   | 60           | ``FEAT_SM3`` && ``FEAT_SM4``        | sm                  | ID_AA64ISAR0_EL1.SM4 == 0b0001 &&   |
+   |              |                                     |                     | ID_AA64ISAR0_EL1.SM3 == 0b0001      |
+   +--------------+-------------------------------------+---------------------+-------------------------------------+
+   | 70           | ``FEAT_RDM``                        | rdm                 | ID_AA64ISAR0_EL1.RDM == 0b0001      |
+   +--------------+-------------------------------------+---------------------+-------------------------------------+
+   | 80           | ``FEAT_LSE``                        | lse                 | ID_AA64ISAR0_EL1.LSE == 0b0001      |
+   +--------------+-------------------------------------+---------------------+-------------------------------------+
+   | 90           | crc32                               | crc32               | ID_AA64ISAR0_EL1.CRC32 == 0b0001    |
+   +--------------+-------------------------------------+---------------------+-------------------------------------+
+   | 100          | ``FEAT_SHA1``                       | sha1                | ID_AA64ISAR0_EL1.LSE == 0b0001      |
+   +--------------+-------------------------------------+---------------------+-------------------------------------+
+   | 110          | ``FEAT_SHA256``                     | sha256              | ID_AA64ISAR0_EL1.SHA2 == 0b0001     |
+   +--------------+-------------------------------------+---------------------+-------------------------------------+
+   | 120          | ``FEAT_SHA512``                     | sha512              | ID_AA64ISAR0_EL1.SHA2 == 0b0010     |
+   +--------------+-------------------------------------+---------------------+-------------------------------------+
+   | 130          | ``FEAT_AES``                        | aes                 | ID_AA64ISAR0_EL1.AES == 0b0001      |
+   +--------------+-------------------------------------+---------------------+-------------------------------------+
+   | 140          | Floating-point                      | fp                  | ID_AA64PFR0_EL1.FP != 0b1111        |
+   +--------------+-------------------------------------+---------------------+-------------------------------------+
+   | 150          | Advanced SIMD                       | advsimd             | ID_AA64PFR0_EL1.AdvSIMD != 0b1111   |
```

Comment:
Do we need `advsimd` and `sve` to be somehow aware of / compatible with the mangling of the names of the Vector Function ABI mangling? https://github.com/ARM-software/abi-aa/blob/main/vfabia64/vfabia64.rst#vector-function-name-mangling

Also, which of the two available calling conventions for NEON is the `advsimd` version supposed to follow? The standard CC or the `aarch64_vector_pcs` one? The latter is more efficient for leaf functions that are passing values via NEON registers. (see https://github.com/ARM-software/abi-aa/blob/main/vfabia64/vfabia64.rst#vector-procedure-call-standard) The

Reply:
I'll add a note, FMV won't change the calling convention because the caller always will call the function in the same way.

Reply:
I have two cases in my mind.

1. What happens if someone defines a NEON function with `aarch64_vector_pcs` and the compiler decides to use multi-versioning on it?
2. What happens if the compiler does multi-versioning of a function that does not use NEON registers in the parameter list, but that gets auto-vectorized to execute in the NEON units?

Reply:
1. The function declaration should have the `__attribute__ ((aarch64_vector_pcs))` and all function definition too. If only one of the versions would uses the vertor pcs then it won't work since the caller has no information about which version will run.  This example should work:
``` 
void __attribute__ ((aarch64_vector_pcs)) foo(uint32x2_t a);
//...
void __attribute__ ((aarch64_vector_pcs ,target_varinant("default")) foo(uint32x2_t a){}
void __attribute__ ((aarch64_vector_pcs ,target_varinant("advsimd")) foo(uint32x2_t a){}
```

2. In this case the function should just should conform to the aapcs as it would do without multi-versioning.
Following example generates two `foo` functions where the sve2 variant may be autovectorised.
```
void __attribute__ ((target_clones("sve2")) foo(){}
```